components/x-interaction/src/Hydrate.jsx

```diff
@@ -1,27 +1,80 @@
-import { h, render } from '@financial-times/x-engine';
+import { h, render, Component } from '@financial-times/x-engine';
 import { getComponent } from './concerns/register-component';
 
-function hydrate() {
+export class HydrationWrapper extends Component {
+	render() {
+		const {Component, props, id} = this.props;
+		return <Component {...props} id={id} actionsRef={a => this.actions = a} />;
+	}
+
+	componentDidMount() {
+		if(this.props.wrapper) {
+			this.props.wrapper.addEventListener('x-interaction.trigger-action', this);
+		}
+	}
+
+	componentWillUnmount() {
+		if(this.props.wrapper) {
+			this.props.wrapper.removeEventListener('x-interaction.trigger-action', this);
+		}
+	}
+
+	handleEvent(event) {
+		const {action, args = []} = event.detail;
+
+		if(this.actions && this.actions[action]) {
+			this.actions[action](...args);
+		}
+	}
+}
+
+export function hydrate() {
 	if (typeof window === 'undefined') {
 		throw new Error('x-interaction hydrate should only be called in the browser');
 	}
 
 	if (!('_xDashInteractionHydrationData' in window)) {
 		throw new Error(
-			"x-interaction hydrate was called without hydration data available. this could happen if you called hydrate before the hydration data was defined, or if you're not ouptutting the hydration data in your server-rendered markup."
+			"x-interaction hydrate was called without hydration data available. this could happen if you called hydrate above where the hydration data was defined on the page, or if you're not ouptutting the hydration data in your server-rendered markup."
```

Comment:
ouptutting => outputting


How about this?
> this can happen if you call hydrate before the serialised data is available or if you're not including the serialised data in your server-rendered markup.